frontend/src/commons/ShareDialog/AllContactsItem.js

```diff
@@ -0,0 +1,171 @@
+    () => records.every(record => record.viewSwitchReadonly),
+    [records]
+  )
+  const shareSwitchReadonly = useMemo(
+    () => records.every(record => record.shareSwitchReadonly),
+    [records]
+  )
+
+  useEffect(() => {
+    const toggleView = records
+      .filter(record => !record.viewSwitchReadonly)
+      .some(record => record.canViewEvent)
+    const toggleShare = records
+      .filter(record => !record.shareSwitchReadonly)
+      .some(record => record.canShareEvent)
+
+    setViewChecked(toggleView)
+    setShareChecked(toggleShare)
+  }, [records, setViewChecked, setShareChecked])
+
+  const switchView = useCallback(() => {
+    if (!viewChecked) {
+      setViewChecked(true);
+      const newRecords = records.map((record) => {
+        if (record.viewSwitchReadonly) {
+          return record;
+        }
+
+        addInvitation(record.describes, ['view']);
+        return { ...record, canViewEvent: true }
+      })
+      setRecords(newRecords)
+    } else {
+      setViewChecked(false);
+      setShareChecked(false);
+      const newRecords = records.map((record) => {
+        if (record.viewSwitchReadonly) {
+          return record
+        }
+
+        removeInvitation(record.describes);
+        return { ...record, canViewEvent: false }
+      })
+      setRecords(newRecords)
+    }
+  }, [
+    records,
+    setRecords,
+    addInvitation,
+    removeInvitation,
+    viewChecked,
+    setViewChecked,
+    setShareChecked
+  ]);
+
+  const switchShare = useCallback(() => {
+    if (!shareChecked) {
+      setShareChecked(true);
+
+      const newRecords = records.map((record) => {
+        if (record.shareSwitchReadonly) {
+          return record;
+        }
+
+        if (record.canViewEvent) {
+          addInvitation(record.describes, ['share']);
+          return { ...record, canShareEvent: true }
+        }
+        else {
+          setViewChecked(true);
+          addInvitation(record.describes, ['view', 'share']);
+          return { ...record, canViewEvent: true, canShareEvent: true }
+        }
+      })
+      setRecords(newRecords)
+    } else {
+      setShareChecked(false);
+
+      const newRecords = records.map((record) => {
+        if (record.shareSwitchReadonly) {
+          return record;
+        }
+        if (record.canViewEvent) {
+          removeInvitation(record.describes);
+          return { ...record, canShareEvent: false }
+        } else {
+          addInvitation(record.describes, ['view']);
+          return { ...record, canViewEvent: true, canShareEvent: false }
+        }
+      })
+      setRecords(newRecords)
+    }
+  }, [
+    records,
+    setRecords,
+    addInvitation,
+    removeInvitation,
+    shareChecked,
+    setViewChecked,
+    setShareChecked
+  ]);
+
+  return (
+    <ListItem className={classes.listItem}>
+      <ListItemAvatar className={classes.avatarItem}>
+        <Avatar className={classes.avatar}>
+          C
+        </Avatar>
```

Comment:
Could you use a [Material-UI icon](https://v4.mui.com/components/material-icons/) to make this line very different from others ? I suggest this one:
`import PublicIcon from '@material-ui/icons/Public';`